cdap-common/src/main/java/io/cdap/cdap/common/operation/LongRunningOperationRequest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.operation;
+
+import io.cdap.cdap.proto.id.OperationRunId;
+import io.cdap.cdap.proto.operationrun.OperationType;
+
+/**
+ * Encapsulates a operation run request.
+ *
+ * @param <T> The type of the operation request.
+ */
+public class LongRunningOperationRequest<T> {
+
+  private final OperationRunId runId;
+  private final OperationType operationType;
+  private final T operationRequest;
```

Comment:
this naming is confusing, there is a request within a request. The generics also seem messy, there is one here, one in LongRunningOperation, and another in OperationRunDetail that are all supposed to match. There should be a cleaner way to model this.

Reply:
I can think of two broad approaches in terms of the generic:

1. Remove the generic entirely and add a field for each operation type (similar to https://cloud.google.com/dataproc/docs/reference/rest/v1/projects.regions.jobs#Job)

2. Keep the generic but make the type part of the OperationId, so that we always know what the corresponding request class for an operation should be. When listing operations of multiple types, either leave out the request entirely or represent it as a JsonObject. The corresponding class for an OperationType would be statically mapped, for example as a field in the actual OperationType enum. 

Haven't thought about this in too much depth, but I feel like the first approach is simpler and less error prone, as we don't need to deserialize things differently depending on an enum value. Since these are all internal classes, we are free to refactor them later if we find there are a lot of different operation types and it is becoming unwieldy.

Reply:
@albertshau Makes sense for the first approach. Union type or `Oneof` would have been the best fit here. Anyway as you suggested the classes can always be changed. 
I will make the necessary changes. I think the only place I would still need generics is the LongRunningOperation interface.